rustls/src/msgs/handshake.rs

```diff
@@ -104,11 +104,9 @@ pub struct SessionID {
 
 impl fmt::Debug for SessionID {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let mut t = f.debug_tuple("SessionID");
-        for i in 0..self.len() {
-            t.field(&self.data[i]);
-        }
-        t.finish()
+        f.debug_tuple("SessionID")
+            .field(&&self.data[..self.len])
```

Comment:
I am surprised at the `&&` instead of `&`. Not sure that's intentional.

Reply:
It is. `field()` takes a `&dyn Debug`, the slice is unsized so it cannot be passed as an argument directly. (Compilation fails without the extra &.)

Reply:
The exact reason here is that only `Sized` types can be cast to `dyn`. Added a comment to that effect.